src/controllers/colormapperjsproxy.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <QScriptContext>
+#include <QScriptEngine>
+
+#include "controllers/colormapper.h"
+
+class ColorMapperJSProxy final : public QObject {
+    Q_OBJECT
+  public:
+    ColorMapperJSProxy() = delete;
+    ColorMapperJSProxy(QScriptEngine* pScriptEngine, QMap<QRgb, QVariant> availableColors);
+
+    ~ColorMapperJSProxy() {
+        delete m_colorMapper;
+    };
+
+  public slots:
```

Comment:
A slot that returns a value other than `void`??

Reply:
For some reason I thought I needed a slot here. Can't find anything in the documentation regarding this, so I don't know how I got that idea. Strange.

Reply:
Whoops, looks like we need it to be a slot with a return value. If I just make it a normal public function, this happens:
![2020-03-09-105216_779x597_scrot](https://user-images.githubusercontent.com/1834516/76201832-2f32e300-61f4-11ea-84f4-87e8fcec5db6.png)

Reply:
Agreed. It seems to be needed for the script binding. It's not a regular slot invoked by the event loop but as a (meta) method. A comment for clarification doesn't hurt.